src/torchcodec/_core/Encoder.cpp

```diff
@@ -0,0 +1,214 @@
+#include "src/torchcodec/_core/Encoder.h"
+#include "torch/types.h"
+
+extern "C" {
+#include <libavcodec/avcodec.h>
```

Comment:
These are already included in `FFMPEGCommon.h`, so we shouldn't need to include them here again. (This inspired #615.)